File: app/src/main/java/com/flyjingfish/lightaop/BaseLightAop.java
package com.flyjingfish.lightaop;

import org.aspectj.lang.ProceedingJoinPoint;

public interface BaseLightAop {
    void beforeInvoke(DebugLog debugLog);
    Object invoke(ProceedingJoinPoint joinPoint,DebugLog debugLog);

    void afterInvoke(DebugLog debugLog);


}

File: light-aop-annotation/src/main/java/com/flyjingfish/light_aop_annotation/BaseLightAop.kt
package com.flyjingfish.light_aop_annotation

import org.aspectj.lang.ProceedingJoinPoint

interface BaseLightAop<T : Annotation?> {
    fun beforeInvoke(annotation: T)
    operator fun invoke(joinPoint: ProceedingJoinPoint, annotation: T): Any?
    fun afterInvoke(annotation: T)
}